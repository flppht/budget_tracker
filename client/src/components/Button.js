import React from "react";
const classnames = require("classnames");

const Button = (props) => {
  const classString = classnames(
    "rounded-full text-white px-2 py-1 align-middle w-1/2 self-center shadow-sm",
    props.className
  );

  const handleClick = () => {
    if (props.onClick) {
      props.onClick();
    }
  };

  return (
    <button type={props.type} className={classString} onClick={handleClick}>
      {props.children}
    </button>
  );
};

export default Button;
