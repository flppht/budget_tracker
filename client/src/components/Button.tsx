const classnames = require("classnames");

type ButtonProps = {
  className?: string;
  type: "submit" | "reset" | "button";
  children: React.ReactNode;
  onClick?: () => void;
  disabled?: boolean;
};

const Button = ({ className, type, children, ...rest }: ButtonProps) => {
  const classString = classnames(
    "rounded-full text-white px-2 py-1 mb-1 align-middle w-1/2 self-center shadow-sm",
    className
  );

  return (
    <button type={type} className={classString} {...rest} data-testid="button">
      {children}
    </button>
  );
};

export default Button;
