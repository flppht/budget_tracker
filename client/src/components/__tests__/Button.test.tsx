import "@testing-library/jest-dom";
import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Button from "../Button";

afterEach(() => {
  cleanup();
});

test("should render Button component", () => {
  render(<Button type="button">OK</Button>);

  const buttonElement = screen.getByTestId("button");

  expect(buttonElement).toBeInTheDocument();
  expect(buttonElement).toHaveTextContent("OK");
});

test("click event works on component", () => {
  const handleClick = jest.fn();

  render(
    <Button type="button" onClick={handleClick}>
      OK
    </Button>
  );

  const buttonElement = screen.getByTestId("button");

  fireEvent.click(buttonElement);

  expect(handleClick).toHaveBeenCalledTimes(1);
});

test("matches snapshot", () => {
  const tree = renderer.create(<Button type="button">OK</Button>).toJSON();

  expect(tree).toMatchSnapshot();
});
