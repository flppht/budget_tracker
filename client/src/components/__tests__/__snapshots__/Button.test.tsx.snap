// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="rounded-full text-white px-2 py-1 mb-1 align-middle w-1/2 self-center shadow-sm"
  data-testid="button"
  type="button"
>
  OK
</button>
`;
